File: choco/choco.rb
include Math

def gcd(n,m)
  if(n > m)
    n,m = m,n
  end
  while n != 0
    n,m = m%n,n
  end
  return m
end

def factorize(n)
  d = 1
  y = rand(n)
  while d == 1
    x = y
    y = rand(n)
    d = gcd((x-y).abs,n)
    if d == n
      return 0
    end
  end
  return d
end

n = 280671392065546467397265294532969672241810318954163887187279320454220348884327

f = factorize(n)

print(f,"x",n/f,"\n")

File: choco/choco.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>
#include <queue>
#include <vector>
#include <cmath>
#include <cstdlib>
#include <ctime>
#include <algorithm>
using namespace std;

int gcd(int n,int m){
  if(n > m) swap(n,m);
  int tmp;  
  while(n != 0){
    tmp = n;
    n = m%n;
    m = tmp;
  }
  return m;
}

int factorize(int n){
  int d = 1;
  int x;
  int y = random()%n+1;
  while(d == 1){
    x = y;
    y = random()%n+1;
    d = gcd(abs(x-y),n);
    if(d == n) return 0;
  }
  return d;
}

int main(){
  srandom(0);
  //int n = 394587;
  int n = 280671392065546467397265294532969672241810318954163887187279320454220348884327;
  int f = factorize(n);
  cout << f << endl;
  cout << n/f << endl;
}
